pinot-plugins/pinot-input-format/pinot-csv/src/main/java/org/apache/pinot/plugin/inputformat/csv/CSVRecordReader.java

```diff
@@ -83,20 +105,34 @@ public void init(File dataFile, @Nullable Set<String> fieldsToRead, @Nullable Re
       }
       char delimiter = config.getDelimiter();
       format = format.withDelimiter(delimiter);
+
+      if (config.isSkipUnParseableLines()) {
+        _useLineIterator = true;
+      }
+
       String csvHeader = config.getHeader();
       if (csvHeader == null) {
         format = format.withHeader();
       } else {
-        //validate header for the delimiter before splitting
-        validateHeaderForDelimiter(delimiter, csvHeader, format);
-        format = format.withHeader(StringUtils.split(csvHeader, delimiter));
+        // do not validate header if using the line iterator
```

Comment:
I think that there's no harm in running `validateHeaderForDelimiter`? It's reading the first line and check the delimiter.

Reply:
Following problems exist with the current validation.
1) It calls `iterator.hasNext()` which is what we are trying to avoid in the first place
2) It checks if the record has multiple values. User can pass in header and the record can have a single line which is valid. Hence, this check is not valid.
3) It checks if delimiter is present in header. This is also not valid if the file is like this.
```
id
100
```

As all these checks are harmful, I have not made use of it.